app/javascript/components/supplemental_application/sections/Lease.js

```diff
@@ -126,18 +127,45 @@ const Lease = ({ form, values, store }) => {
     ...map(confirmedPreferences, pluck('id', 'preference_name'))
   ])
 
-  const getVisited = (fieldName) => form.getFieldState(fieldName)?.visited
+  const availableUnits = units.filter(
+    (unit) => !unit.application_id || unit.application_id === application.id
+  )
 
-  const openDeleteLeaseConfirmation = () => setShowDeleteConfirmation(true)
-  const closeDeleteLeaseConfirmation = () => setShowDeleteConfirmation(false)
+  const availableUnitsOptions = formUtils.toOptions(
+    map(availableUnits, pluck('id', 'unit_number', 'priority_type'))
+  )
 
-  const handleDeleteLeaseConfirmed = () => {
-    setShowDeleteConfirmation(false)
-    handleDeleteLease()
-  }
+  const selectedUnit = form.getState().values.lease?.unit
 
   const areNoUnitsAvailable = !availableUnitsOptions.length
 
+  const prefUsedIs = (pref) => {
+    const prefUsed = confirmedPreferences.find(
+      (pref) => pref.id === form.getState().values.lease?.preference_used
+    )
+    return prefUsed?.preference_name.includes(pref)
```

Comment:
nit: `prefUsed?.preference_name.includes(pref)` makes it seem like you want `preference_name` to be nullable, but then the includes call will fail. Depending on if you think it should be nullable, make the whole chain use `?.` or `.`

Also: is there a reason for using includes instead of `===`? Is there ever a case where `pref` is expected to be a substring of `preference_name`?

Reply:
Mostly just laziness wanting to have the strings be "DTHP" and "NRHP" instead of "Displaced Tenant Housing Preference (DTHP)" but since I have a constant now maybe that makes more sense to just do the equality check